pkg/virt-launcher/virtwrap/access-credentials/access_credentials.go

```diff
@@ -306,7 +309,26 @@ func (l *AccessCredentialManager) pingAgent(domName string) error {
 	return err
 }
 
-func (l *AccessCredentialManager) agentWriteAuthorizedKeys(domName string, user string, desiredAuthorizedKeys string) error {
+func (l *AccessCredentialManager) agentSetAuthorizedKeys(domName string, user string, authorizedKeys []string) error {
+	err := func() error {
+		domain, err := l.virConn.LookupDomainByName(domName)
+		if err != nil {
+			return err
+		}
+		defer domain.Free()
+
+		// Zero flags argument means that the authorized_keys file is overwritten with the authorizedKeys
+		return domain.AuthorizedSSHKeysSet(user, authorizedKeys, 0)
+	}()
+	if errors.Is(err, libvirt.ERR_NO_SUPPORT) {
+		// If AuthorizedSSHKeysSet method is not supported, use the old method
+		desiredAuthorizedKeys := strings.Join(authorizedKeys, "\n")
+		return l.agentWriteAuthorizedKeysFile(domName, user, desiredAuthorizedKeys)
+	}
+	return err
+}
+
+func (l *AccessCredentialManager) agentWriteAuthorizedKeysFile(domName string, user string, desiredAuthorizedKeys string) (err error) {
```

Comment:
Maybe you want to mark this as deprecated and/or also include "legacy" or a similar word in the name.

Reply:
If I can assume that we use correct libvirt version, I can remove this function.